From 5ea46ea239b86e1482673114f98540effabd9aeb Mon Sep 17 00:00:00 2001
From: Pankaj Dev <pankaj.dev@st.com>
Date: Tue, 23 Jan 2024 17:56:39 +0530
Subject: [PATCH 0922/1141] arm64: dts: st: Add wakeup source for usb3dr using
 exti on stm32mp25

Remote-wakeup capable device connected to USB3DR can act as a
source to wakeup system from low-power state on mp25 platform
Enable exti-intr connected to utmi_suspend_n and utmi_l1_suspend_n
output of USB3DR as wakeup-source

Change-Id: I294dc157a73eb47e8563ccb06ff9007dd36a0b11
Signed-off-by: Pankaj Dev <pankaj.dev@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/353692
Reviewed-by: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Fabrice GASNIER <fabrice.gasnier@foss.st.com>
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp251.dtsi | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp251.dtsi b/arch/arm64/boot/dts/st/stm32mp251.dtsi
index 05f19148a88c..6d9d8e84c77f 100644
--- a/arch/arm64/boot/dts/st/stm32mp251.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp251.dtsi
@@ -2274,6 +2274,8 @@ usb3dr: usb@48300000 {
 				ranges = <0x48300000 0x48300000 0x100000>;
 				feature-domains = <&rifsc STM32MP25_RIFSC_USB3DR_ID>;
 				power-domains = <&CLUSTER_PD>;
+				wakeup-source;
+				interrupts-extended = <&exti1 44 IRQ_TYPE_EDGE_RISING>;
 				status = "disabled";
 
 				dwc3: usb@48300000 {
@@ -2286,6 +2288,7 @@ dwc3: usb@48300000 {
 					resets = <&rcc USB3DR_R>;
 					phys = <&usb2_phy2>;
 					phy-names = "usb2-phy";
+					wakeup-source;
 				};
 			};
 
-- 
2.39.2

